src/Ocelot.Provider.Kubernetes/Kube.cs

```diff
@@ -46,9 +48,42 @@ public virtual async Task<List<Service>> GetAsync()
             .ToList();
     }
 
-    private Task<EndpointsV1> GetEndpoint() => _kubeApi
-        .ResourceClient<IEndPointClient>(client => new EndPointClientV1(client))
-        .GetAsync(_configuration.KeyOfServiceInK8s, _configuration.KubeNamespace);
+    private async Task<EndpointsV1> GetEndpoint()
```

Comment:
I've moved this logic here because it keeps the resource-client implementation cleaner (since it wouldn't otherwise be clear to someone who's not seen this code before what the lifetime of the resource client is, vs the lifetime of the containing type or its logger).